cluster/src/main/scala/org/apache/spark/memory/SnappyUnifiedMemoryManager.scala

```diff
@@ -385,6 +385,15 @@ private object SnappyUnifiedMemoryManager extends Logging{
     testCallbacks.map(l => l.onPositiveMemoryIncreaseDueToEviction(objectName, bytes))
   }
 
+  def getMemorySize(conf: SparkConf): Long = {
+    val cache = Misc.getGemFireCacheNoThrow
+    if (cache ne null) {
+      OffHeapStorage.parseOffHeapMemorySize(
+        Misc.getGemFireCache.getDistributedSystem.getConfig.getMemorySize)
```

Comment:
@rishitesh Can you add this long value as a field+getter in GemFireCacheImpl constructor itself? Then it can be used both here as well as inside GemFireCacheImpl for initializing BufferAllocator used for off-heap allocation (in SNAP-1454)